### PR TITLE
[Perf-Mon] Enable Performance Monitoring

### PR DESCRIPTION
**Pull request by pranavprakash20**
This PR contains added support for monitoring performance and CPU usage of processes. To enable this feature, simply add `--monitor-performance` tag in the run.py command. This will enable the monitoring of processes defined by the user in `~\.cephci.yaml` on the specific nodes for each test. Then the results are copied to the log-dir upon completion of the run

Add the following into the ~/.cephci.yaml

"""

performace_monitoring:
  processes_to_monitor: ceph-osd, ceph-mgr, ceph-mon, ceph-mds
  interval: 10

""" 

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/perfmon-run/performance-metrics/ => Perf data
http://magna002.ceph.redhat.com/cephci-jenkins/perfmon-run/ => Run log

**Review by MohitBis — APPROVED** [review]
LGTM

**Comment by pranavprakash20** [comment]
NFS Ganesha specific per data for RHCEPH 7.0 => http://magna002.ceph.redhat.com/cephci-jenkins/nfs-7-0-tier-1-phase-2-perf-data/performance-metrics/

File: run.py [hunk 1]
```diff
@@ -685,6 +696,17 @@ def fetch_test_details(var) -> dict:
         "log_dir": run_dir,
         "run_id": run_id,
     }
+    download_path = run_dir if not log_directory else log_directory
+
+    # If Performance and CPU usage monitoring is enabled, perform pre-reqs
+    if enable_perf_mon:
```
Comment (vamahaja) [inline run.py:702]: This won't work in case we have multisite or mirror cluster.

Comment (pranavprakash20) [reply run.py]: Noted. The logic has been added under the exisitng for loop that iterates over the clusters

**Comment by pranavprakash20** [comment]
new run log with changes made : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-HTHDF1/performance-metrics/

**Review by vamahaja — APPROVED** [review]

**Pull request merged**